### PR TITLE
test: Python models filtering outputs based on requested outputs

### PR DESCRIPTION
**Pull request by kthui**
#### What does the PR do?
Add tests to establish the behavior of Python backend filtering outputs based on requested outputs from the client.

#### Checklist
- [x] PR title reflects the change and is of format `<commit_type>: <Title>`
- [x] Changes are described in the pull request.
- [x] Related issues are referenced.
- [x] Populated [github labels](https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels) field
- [x] Added [test plan](#test-plan) and verified test passes.
- [x] Verified that the PR passes existing CI.
- [x] Verified copyright is correct on all changed files.
- [ ] Added _succinct_ git squash message before merging [ref](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).
- [x] All template sections are filled out.
- [ ] Optional: Additional screenshots for behavior/output changes with before/after.

#### Commit Type:
Check the [conventional commit type](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type)
box here and add the label to the github PR.
- [ ] build
- [ ] ci
- [ ] docs
- [ ] feat
- [ ] fix
- [ ] perf
- [ ] refactor
- [ ] revert
- [ ] style
- [x] test

#### Related PRs:
https://github.com/triton-inference-server/python_backend/pull/366

#### Where should the reviewer start?
N/A

#### Test plan:
N/A

- CI Pipeline ID:
#15726534
#15745686
#15752015

#### Caveats:
N/A

#### Background
N/A

#### Related Issues: (use one of the action keywords Closes / Fixes / Resolves / Relates to)
N/A

### CODE REVIEW

File: qa/L0_backend_python/io/io_test.py [hunk 1]
```diff
@@ -162,6 +162,80 @@ def test_variable_gpu_output(self):
...
+                "GPU_OUTPUT",
+                gpu_output_data.shape,
+                np_to_triton_dtype(gpu_output_data.dtype),
+            ),
+        ]
+        inputs[0].set_data_from_numpy(input0_data)
+        inputs[1].set_data_from_numpy(gpu_output_data)
+
+        # request for output 0, among output 0 and next gpu output.
+        requested_outputs = [grpcclient.InferRequestedOutput("OUTPUT0")]
```
Comment (Tabrizian) [inline qa/L0_backend_python/io/io_test.py:219]: If we already have the test and just removing that condition would fix decoupled mode as well, let's update the test to make it work for decoupled mode as well.

Comment (kthui) [reply qa/L0_backend_python/io/io_test.py:219]: Sure, [Reverse decoupled requested output filtering test goal](https://github.com/triton-inference-server/server/pull/7338/commits/1388172906cd2844588402fa20a282b4959591ff)

**Review by rmccorm4 — APPROVED** [review]
LGTM :rocket: ~~Also need a sanity check that `L0_batcher` tests pass with the python_backend change as well~~ Confirmed L0_batcher passes same pipeline.

File: qa/L0_backend_python/io/io_test.py (hunk 1 above)
Comment (GuanLuo) [reply qa/L0_backend_python/io/io_test.py:219]: We probably have tested that in other test already, do we have test to make sure "not specifying requested output returns all outputs"?

**Review by Tabrizian — DISMISSED** [review]

File: qa/L0_backend_python/io/io_test.py (hunk 1 above)
Comment (kthui) [reply qa/L0_backend_python/io/io_test.py:228]: We do have it in other tests, but not explicitly test for it. It is an easy addition into the existing two tests, for a complete testing of the requested output behavior: [Add verification for all outputs returned without requested output](https://github.com/triton-inference-server/server/pull/7338/commits/b6e10f82b29840fe5700824fc8072db1edc123f1)

**Review by rmccorm4 — APPROVED** [review]

**Pull request merged**